include/wasm/WasmModule.h

```diff
@@ -159,12 +168,10 @@ class WasmModule
     uint32_t threadPoolSize = 0;
     std::vector<uint32_t> threadStacks;
 
-    faabric::util::Queue<std::pair<std::promise<int32_t>, threads::OpenMPTask>>
-      openMPTaskQueue;
-    faabric::util::Queue<std::pair<std::promise<int32_t>, threads::PthreadTask>>
-      pthreadTaskQueue;
+    OpenMPQueueMap openMPTaskQueueMap;
+    PthreadQueue pthreadTaskQueue;
 
-    std::vector<std::thread> openMPThreads;
+    std::unordered_map<int, std::thread> openMPThreads;
```

Comment:
We care about which thread is which, and may not need all of them so a map seemed to make sense. It could also be an array of pointers, which is what we use for the thread contexts in the `WAVMModule`. Given that this works and doesn't seem to be a performance issue I'm inclined to leave that for later.